spec/lib/salus/scanners/slither_spec.rb

```diff
@@ -0,0 +1,153 @@
+      repo = Salus::Repo.new('spec/fixtures/slither/solidity-good')
+      scanner = Salus::Scanners::Slither.new(repository: repo, config: {})
+      expect(scanner).not_to receive(:report_failure)
+      scanner.run
+      expect(scanner.report.to_h.fetch(:passed)).to eq(true)
+    end
+
+    it 'should fail when there are vulnerabilities' do
+      repo = Salus::Repo.new('spec/fixtures/slither/solidity-bad')
+      scanner = Salus::Scanners::Slither.new(repository: repo, config: {})
+      scanner.run
+
+      expect(scanner.report.to_h.fetch(:passed)).to eq(false)
+      stdout = scanner.report.to_h[:info][:stdout]
+      stdout = JSON.parse(stdout)
+      expected_ref_url = "https://github.com/crytic/slither/wiki/Detector-Documentation#incorrect-shift"
+      expected_vul1 = { "description" => "C.f() (bad-contract.sol#4-8) contains an incorrect "\
+                                         "shift operation: a = 8 >> a (bad-contract.sol#6)\n",
+                       "location" => "contracts/bad-contract.sol#L4-L8",
+                       "check" => "incorrect-shift",
+                       "ref_url" => expected_ref_url,
+                       "impact" => "High",
+                       "confidence" => "High" }
+      expected_vul2 = { "description" => "C.g() (bad-contract.sol#10-14) contains an incorrect "\
+                                         "shift operation: b = 8 >> b (bad-contract.sol#12)\n",
+                       "location" => "contracts/bad-contract.sol#L10-L14",
+                       "check" => "incorrect-shift",
+                       "ref_url" => expected_ref_url,
+                       "impact" => "High",
+                       "confidence" => "High" }
+      expect(stdout.length).to eq(2)
+      expect(stdout).to include(expected_vul1)
+      expect(stdout).to include(expected_vul2)
+    end
+
+    it 'should report error if solidity code has invalid syntax' do
+      repo = Salus::Repo.new('spec/fixtures/slither/no-compile-truffle')
+      scanner = Salus::Scanners::Slither.new(repository: repo, config: {})
+      scanner.run
+
+      expect(scanner.report.to_h.fetch(:passed)).to eq(false)
+      logs = scanner.report.to_h[:logs]
+      expect(logs).to eq(nil)
+      errs = scanner.report.to_h.fetch(:errors)
+      expect(errs.size).to eq(1)
+      expect(errs[0][:message]).to include('truffle compile` failed. Can you run it?')
+    end
+  end
+
+  describe '#version_valid?' do
+    context 'scanner version is valid' do
+      it 'should return true' do
+        repo = Salus::Repo.new('spec/fixtures/blank_repository')
+        scanner = Salus::Scanners::Slither.new(repository: repo, config: {})
+        expect(scanner.version).to be_a_valid_version
+      end
+    end
+  end
+
+  describe '#supported_languages' do
+    context 'should return supported languages' do
+      it 'should return solidity' do
+        langs = Salus::Scanners::Slither.supported_languages
+        expect(langs).to eq(['solidity'])
```

Comment:
expect(described_class.supported_languages).to eq(['solidity'])

Reply:
updated accordingly